samplers/src/test/java/internal/CelBasedSamplerComponentProviderTest.java

```diff
@@ -0,0 +1,244 @@
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class CelBasedSamplerComponentProviderTest {
+
+  private static final CelBasedSamplerComponentProvider PROVIDER =
+      new CelBasedSamplerComponentProvider();
+
+  @Test
+  void endToEnd() {
+    String yaml =
+        "file_format: 0.4\n"
+            + "tracer_provider:\n"
+            + "  sampler:\n"
+            + "    parent_based:\n"
+            + "      root:\n"
+            + "        cel_based:\n"
+            + "          fallback_sampler:\n"
+            + "            always_on:\n"
+            + "          expressions:\n"
+            + "            - action: DROP\n"
+            + "              expression: '\"example.com\" in attribute[\"http.response.header.host\"] && attribute[\"http.response.status_code\"] == 200'\n"
+            + "            - action: DROP\n"
+            + "              expression: spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")\n"
+            + "            - action: RECORD_AND_SAMPLE\n"
+            + "              expression: spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")\n";
+    OpenTelemetrySdk openTelemetrySdk =
+        DeclarativeConfiguration.parseAndCreate(
+            new ByteArrayInputStream(yaml.getBytes(StandardCharsets.UTF_8)));
+    Sampler sampler = openTelemetrySdk.getSdkTracerProvider().getSampler();
+    assertThat(sampler.toString())
+        .isEqualTo(
+            "ParentBased{"
+                + "root:CelBasedSampler{"
+                + "fallback=AlwaysOnSampler, "
+                + "expressions=["
+                + "CelBasedSamplingExpression{"
+                + "expression='\"example.com\" in attribute[\"http.response.header.host\"] && attribute[\"http.response.status_code\"] == 200', "
+                + "delegate=AlwaysOffSampler"
+                + "}, "
+                + "CelBasedSamplingExpression{"
+                + "expression='spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")', "
+                + "delegate=AlwaysOffSampler"
+                + "}, "
+                + "CelBasedSamplingExpression{"
+                + "expression='spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")', "
+                + "delegate=AlwaysOnSampler"
+                + "}"
+                + "]},"
+                + "remoteParentSampled:AlwaysOnSampler,"
+                + "remoteParentNotSampled:AlwaysOffSampler,"
+                + "localParentSampled:AlwaysOnSampler,"
+                + "localParentNotSampled:AlwaysOffSampler"
+                + "}");
+
+    // SERVER span to other path should be recorded and sampled
```

Comment:
I think this one matches the first expression?
```suggestion
    // SERVER span with host header and 200 response is dropped due to matching with first expression.
```